### ISSUE TITLE
JITServer : cmdLineTester_jvmtitests_hcr_OSRG & others - Segmentation error vmState=0x00000000 at libj9jit29.so

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse/openj9#11903]
Failure link
------------

From internal builds: `Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly/303`, `Test_openjdk16_j9_extended.functional_x86-64_linux_jit_Nightly/24`:
```
23:44:40  openjdk version "1.8.0_292-internal"
23:44:40  OpenJDK Runtime Environment (build 1.8.0_292-internal-jenkins_2021_02_06_22_41-b00)
23:44:40  Eclipse OpenJ9 VM (build master-eea5afd5461, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20210206_315 (JIT enabled, AOT enabled)
23:44:40  OpenJ9   - eea5afd5461
23:44:40  OMR      - adc01bb0330
23:44:40  JCL      - c94b45c5cef based on jdk8u292-b01)
```

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
===============================================
Running test cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4 ...
===============================================
cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4 Start Time: Sat Feb  6 23:02:39 2021 Epoch Time (ms): 1612681359930
variation: Mode612-OSRG
JVM_OPTIONS: -XX:+UseJITServer -Xcompressedrefs -Xgcpolicy:gencon -Xjit:enableOSR,enableOSROnGuardFailure,count=1,disableAsyncCompilation 
JITServer is currently a technology preview. Its use is not yet supported
Parsing exclude list...
  add excludes for [all]
  add excludes for [linux_x86-64_cr]
DONE in 0 ms.
Testing: rc001
Test start time: 2021/02/06 23:02:52 Pacific Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdkbinary/j2sdk-image/bin/java" -XX:+UseJITServer -Xcompressedrefs -Xgcpolicy:gencon -Xjit:enableOSR,enableOSROnGuardFailure,count=1,disableAsyncCompilation  -Xdump    -agentlib:jvmtitest=test:rc001 -cp "/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar:/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/../TKG/lib/asm-all.jar" com.ibm.jvmti.tests.util.TestRunner
Time spent starting: 215 milliseconds
Time spent executing: 1421 milliseconds
Test result: FAILED
Output from test:
 [ERR] JITServer is currently a technology preview. Its use is not yet supported
 [ERR] Unhandled exception
 [ERR] Type=Segmentation error vmState=0x00000000
 [ERR] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
 [ERR] Handler1=00007FCE3F274D40 Handler2=00007FCE3EFDAA10 InaccessibleAddress=00007FC85F4D1570
 [ERR] RDI=00007FCE217D2A78 RSI=0000000000000000 RAX=00007FC85F4D156C RBX=0000000000016700
 [ERR] RCX=0000000000000001 RDX=00007FCE2341FF9D R8=00007FCE217D2B61 R9=00007FCE217D2BA0
 [ERR] R10=0000000000000006 R11=000000000000000A R12=00007FCE44E39EF0 R13=00007FCE2341FF9D
 [ERR] R14=00007FCE40013660 R15=00007FCE217D2A78
 [ERR] RIP=00007FCE373F2BBD GS=0000 FS=0000 RSP=00007FCE44E39E40
 [ERR] EFlags=0000000000010246 CS=0033 RBP=00007FCE44638360 ERR=0000000000000004
 [ERR] TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=00007FC85F4D1570
 [ERR] xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm1 2f6176616a4c293b (f: 1783376128.000000, d: 1.840923e-80)
 [ERR] xmm2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm3 000000003eb1353d (f: 1051800896.000000, d: 5.196587e-315)
 [ERR] xmm4 00000000be2f903d (f: 3190788096.000000, d: 1.576459e-314)
 [ERR] xmm5 00000000395a743c (f: 962229312.000000, d: 4.754044e-315)
 [ERR] xmm6 00000000000505a0 (f: 329120.000000, d: 1.626069e-318)
 [ERR] xmm7 0000000000050560 (f: 329056.000000, d: 1.625753e-318)
 [ERR] xmm8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm9 000000003e17cee7 (f: 1041747712.000000, d: 5.146917e-315)
 [ERR] xmm10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm11 0000ff0000000000 (f: 0.000000, d: 1.385239e-309)
 [ERR] xmm12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] xmm15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
 [ERR] Module=/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdkbinary/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
 [ERR] Module_base_address=00007FCE36F85000
 [ERR] Target=2_90_20210206_315 (Linux 3.10.0-1160.11.1.el7.x86_64)
 [ERR] CPU=amd64 (4 logical CPUs) (0x1e8d10000 RAM)
 [ERR] ----------- Stack Backtrace -----------
 [ERR] (0x00007FCE373F2BBD [libj9jit29.so+0x46dbbd])
 [ERR] (0x00007FCE378D725B [libj9jit29.so+0x95225b])
 [ERR] (0x00007FCE378CF637 [libj9jit29.so+0x94a637])
 [ERR] (0x00007FCE378E59A6 [libj9jit29.so+0x9609a6])
 [ERR] ---------------------------------------
 [ERR] JVMDUMP039I Processing dump event "gpf", detail "" at 2021/02/06 23:02:53 - please wait.
 [ERR] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/core.20210206.230253.8681.0001.dmp' in response to an event
 [ERR] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/core.20210206.230253.8681.0001.dmp
 [ERR] JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/javacore.20210206.230253.8681.0002.txt' in response to an event
 [ERR] JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/javacore.20210206.230253.8681.0002.txt
 [ERR] JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/Snap.20210206.230253.8681.0003.trc' in response to an event
 [ERR] JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/Snap.20210206.230253.8681.0003.trc
 [ERR] JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/jitdump.20210206.230253.8681.0004.dmp' in response to an event
 [ERR] JVMDUMP010I JIT dump written to /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_jit_Nightly_testList_1/openjdk-tests/TKG/output_16126788409668/cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4/jitdump.20210206.230253.8681.0004.dmp
 [ERR] JVMDUMP013I Processed dump event "gpf", detail "".
>> Success condition was not found: [Return code: 0]

cmdLineTester_jvmtitests_hcr_OSRG_nongold_SE80_4_FAILED

```
Similar segmentation errors observed in other `cmdLineTester_decompilationTests` & `jit_hw` across JDK `8/11/16`.

**Comment by pshipton** [comment]
@mpirvu fyi. Seems a regression. Note "Similar segmentation errors observed in other cmdLineTester_decompilationTests & jit_hw across JDK 8/11/16." This problem doesn't seem to be occurring in the open builds, not sure why it would only occur in internal testing.

**Comment by mpirvu** [comment]
@dmitry-ten Could you please look into this failures? If you manage to reproduce it we can try to back out the few items we delivered in the past week or so. 
Also of interest is https://github.com/eclipse/openj9/pull/11553
Thanks

**Comment by dmitry-ten** [comment]
Ok, I found the error:
https://github.com/eclipse/openj9/blob/eea5afd5461b62899b47e11eb3404e8a3835f78c/runtime/compiler/runtime/RelocationRuntime.cpp#L753-L761

#11553 added relocation of OSR info so in JITServer it gets relocated twice, resulting in an invalid address.
I'm deleting the second one in my PR #11898

**Comment by mpirvu** [comment]
Good find!. I'll review your PR as soon as possible.

**Comment by mpirvu** [comment]
This issue should be fixed by #11898 which I just merged
